src/alpaca_parser.yrl

```diff
@@ -145,12 +150,36 @@ record_type_members -> record_type_member ',' record_type_members : ['$1' | '$3'
 record_type -> open_brace record_type_members close_brace : 
   #t_record{members='$2'}.
 
+module_qualified_type_name -> symbol '.' symbol:
+  {symbol, L, Mod} = '$1',
+  {symbol, _, Name} = '$3',
+  {module_qualified_type_name, L, Mod, Name}.
+
+module_qualified_type -> module_qualified_type_name :
+  {module_qualified_type_name, L, Mod, Name} = '$1',
+  #alpaca_type{
+     line = L,
+     module = list_to_atom(Mod),
+     name = {type_name, L, Name}}.
+
+module_qualified_type -> module_qualified_type_name type_expressions:
+  {module_qualified_type_name, L, Mod, Name} = '$1',
+  #alpaca_type{
+     line = L,
+     module = list_to_atom(Mod),
+     name = {type_name, L, Name},
+     vars = [V || {type_var, _, _}=V <- '$2']}.
```

Comment:
`int` would not necessarily be a member here though, it would be the value assigned to a type variable in `option.option` and we don't actually have any information about what `option.option`'s full membership is at this point - I think the typer needs to instantiate those.  E.g. the following works without requiring the members to be added in the parser (and I'll add this to a test):

    module m
    export_type t
    type t 'a = T 'a

    module n
    type u 'a = U m.t 'a
    let f () = U m.T 1

But in looking at this I did find another bug!  This does _not_ work, it tries to unify `t_int` with `undefined`:

    module n
    type u = U m.t int
    let f () = U m.T 1

And interestingly this exhibits the exact same unification failure:

    module n
    type opt 'a = Some 'a | None
    type u = U opt int
    let f () = U Some 1

Adding the members in the parser stage does not seem to correct this in the latter example and it seems to me that neither of those should fail.  I haven't dug into this yet but I expect what's happening is that when we do:

    type option 'a = Some 'a | None
    type something_else = option int

The parser has no idea that `int` is supposed to be assigned to a variable `'a` in that second line and so trying to get that variable from the proplist in an `#adt{}` yields `undefined`.  That's a big guess though as I haven't looked too closely.  I'll file this as a new issue.

I suspect this might be as simple as the parser manufacturing variable names for any type parameters that are not explicitly type variables.